tests/benchmarks/multicore/idle_gpio/src/main.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/pm/pm.h>
+
+
+LOG_MODULE_REGISTER(idle_gpio);
+
+#if IS_ENABLED(CONFIG_SOC_NRF54H20_CPUAPP)
+static const struct gpio_dt_spec sw = GPIO_DT_SPEC_GET(DT_ALIAS(sw0), gpios);
+#elif IS_ENABLED(CONFIG_SOC_NRF54H20_CPURAD)
+static const struct gpio_dt_spec sw = GPIO_DT_SPEC_GET(DT_ALIAS(sw1), gpios);
+#else
+#error "Invalid core selected. "
+#endif
+
+int main(void)
+{
+	unsigned int cnt = 0;
+	int rc;
+
+	rc = gpio_pin_configure_dt(&sw, GPIO_INPUT);
+	if (rc < 0) {
+		printf("Could not configure sw GPIO (%d)\n", rc);
+		return 0;
+	}
+	rc = gpio_pin_interrupt_configure_dt(&sw, GPIO_INT_LEVEL_ACTIVE);
+	if (rc < 0) {
+		printf("Could not configure sw GPIO interrupt (%d)\n", rc);
+		return 0;
+	}
+	LOG_INF("Multicore idle_gpio test on %s", CONFIG_BOARD_TARGET);
+	while (1) {
+		LOG_INF("Multicore idle_gpio test iteration %u", cnt++);
+#if IS_ENABLED(CONFIG_SOC_NRF54H20_CPUAPP)
+		pm_state_set(PM_STATE_SUSPEND_TO_RAM, 0);
+#elif IS_ENABLED(CONFIG_SOC_NRF54H20_CPURAD)
+		pm_state_set(PM_STATE_RUNTIME_IDLE, 0);
+#endif
+		k_busy_wait(1000000);
```

Comment:
what about k_sleep and idle state?

Reply:
if you call k_sleep here, the gpio will wake up the CPU, but the scheduler will force sleeping again. We are introducing sleep state by  pm_state_set() while k_busy_wait() is simulating 1s of active state of the CPU